modules/document_repository/jsx/editCategoryForm.js

```diff
@@ -0,0 +1,191 @@
+import PropTypes from 'prop-types';
+import Loader from 'Loader';
+/**
+ * Document Edit Form
+ *
+ * Fetches data from Loris backend and displays a form allowing
+ * to edit a category
+ *
+ * @author Pierre PAC SOO
+ * @version 1.0.0
+ *
+ * */
+class EditDocCategoryForm extends React.Component {
+  /**
+   * @constructor
+   * @param {object} props - React Component properties
+   */
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {},
+      formData: {},
+      uploadResult: null,
+      errorMessage: null,
+      isLoaded: false,
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.fetchData = this.fetchData.bind(this);
+  }
+
+  /**
+   * Called by React when the component has been rendered on the page.
+   */
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  /**
+   * Fetch data
+   * @return {Promise<void>}
+   */
+  fetchData() {
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
```

Comment:
So from the dicussion in [PR-7385](https://github.com/aces/Loris/pull/7385#discussion_r657182678), fetch is very limited in error handling and does not catch errors from the BE. You'll need to add custom error handling in the first `then` to handle.

Reply:
Im not sure I know what to do here.. Can you give me an example of error handling that s similar please?

Reply:
So in general, we've been adding inline validation so you code would become something like
```
fetch(...)
    .then((resp) => resp.json())
    .then((data) => {
        if (!response.ok) {
            // handle error
        }

        // Handle success
    }
```

See the discussion on [PR-7385](https://github.com/aces/Loris/pull/7385#discussion_r657182678)